### PR TITLE
Aurora: Vietnam Edition

### PR DESCRIPTION
**Pull request by FluffyGhoster**
https://youtu.be/s8z6cbThnQQ?si=oVtSp6GObnSylcS5

Added get_turfs_in_cone(), to acquire turfs within a cone.
Added napalm grenades, not as classy as an airstrike, but still pretty effective all the same.
Added punji traps, gives a severe infection upon falling in one, can be hidden, can be engraved with a message.
Added standstill mines, if walked over, they activate and wait for the victim to move out of them, can be disarmed with the bomb kit, pretty powerful explosion if you mess up.
Added claymore mines, project pellets in a 60 degrees radius on each side from the direction they're aiming at, remotely activated via a signaler or by directly touching them.
Reworked flamethrowers, now they light up on fire the turfs in a 30 degrees cone from each side of the clicked target, pressure effects excluded.
Added the napalm grenades, punji traps, standstill mines, claymore mines and the flamethrower to the antag uplink.
DMDoced some variables for the uplink.
DMDoced some variables for the pellet bullets, and minor refactor of its attack_mob function.
Refactored how mines work, added some documentation and sectioned them up to be more clear what is what.
Moved landmines and traps sprites in a more congenial location.
Fixed a runtime error when the uplink UI is opened without a category being selected.
Fixed a runtime error with borgs being destroyed and their HUD refreshing.
Added sprites for the punji trap, by Tomixcomics.
Added sprites for the standstill mine, by Tomixcomics.
Added sprites for the claymore mine, from CM.

### CODE REVIEW

**Comment by FluffyGhoster** [comment]
!review

File: code/datums/uplink/grenades.dm [hunk 1]
```diff
@@ -39,3 +39,9 @@
 	telecrystal_cost = 2
 	path = /obj/item/storage/box/cardox
 	desc = "A box of five grenades that deploy cardox smoke in the thrown area. This smoke is incredibly toxic, especially to vaurca. It can also clear K'ois outbreaks with ease."
+
+/datum/uplink_item/item/grenades/napalm
+	name = "5 x Napalm Grenades"
```
Comment (NonQueueingMatt) [inline code/datums/uplink/grenades.dm:44]:
```suggestion
	name = "5x Napalm Grenades"
```

File: code/game/objects/items/weapons/flamethrower.dm [hunk 2]
```diff
@@ -100,7 +99,8 @@
 	if(user && user.get_active_hand() == src)
 		var/turf/target_turf = get_turf(target)
 		if(target_turf)
-			var/turflist = getline(user, target_turf)
+			//var/turflist = getline(user, target_turf)
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/flamethrower.dm:102]:
```suggestion
```

File: code/game/objects/items/weapons/landmines.dm [hunk 3]
```diff
@@ -171,10 +243,164 @@
...
+	var/engaged_by = null
+
+/obj/item/landmine/standstill/Destroy()
+	STOP_PROCESSING(SSfast_process, src)
+	engaged_by = null
+	. = ..()
+
+/obj/item/landmine/standstill/trigger(mob/living/triggerer)
+	if(!engaged_by && !deactivated)
+		to_chat(triggerer, SPAN_HIGHDANGER("Something clicks below your feet, a sense of dread permeates your skin, better not move..."))
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/landmines.dm:278]: You should probably give the mob some pain and a stun to make it more obvious, otherwise people will just keep walking through in case it's placed under a door or somewhere more conspicuous.

File: code/game/objects/items/weapons/landmines.dm [hunk 4]
```diff
@@ -171,10 +243,164 @@
...
+		engaged_by = ref(triggerer)
+
+		//Because mobs can bump and swap with one another, and we use forcemove that doesn't call Crossed/Uncrossed/Entered/Exited, we have to
+		//keep looking if the victim is still on the mine, and otherwise explode the mine
+		START_PROCESSING(SSfast_process, src)
+		INVOKE_ASYNC(GLOBAL_PROC, GLOBAL_PROC_REF(tgui_alert), triggerer, "You get a dreadful feeling from your [pick("right", "left")] foot...", "Dread", list("Mom..."))
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/landmines.dm:284]: this message isn't very clear, I'm not sure what it's supposed to convey

File: code/game/objects/items/weapons/landmines.dm [hunk 5]
```diff
@@ -171,10 +243,164 @@
...
+
+/**
+ * Called when the actual explosion needs to be performed
+ *
+ * * victim - The `/mob` that triggers the explosion
+ */
+/obj/item/landmine/standstill/proc/late_trigger(mob/living/victim)
+	if(!deactivated)
+		for(var/mob/living/person_in_range in range(world.view, src))
+			to_chat(person_in_range, SPAN_HIGHDANGER("[victim] does a sudden move, releasing the feet from the trigger..."))
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/landmines.dm:319]: use the spatial grid helper here

File: code/game/objects/items/weapons/landmines.dm [hunk 6]
```diff
@@ -171,10 +243,164 @@
...
+/obj/item/landmine/claymore
+	name = "Claymore Landmine"
+	desc = "A landmine that projects sharpnels in a cone of explosion, towards one direction."
+	desc_extended = "A household name, this mine finds extensive use amongst military forces due to its ability to provide area penetration denial and aid ambushes. \
+	It is narrated that Gadpathur, its largest manufacturer in modern times, have built more of these mines than the census of the core planets of the Solarian Alliance."
+	desc_info = "This device can be fitted with a signaler device for remotely actuated detonations, or can be activated with the press of a button directly above it."
+	icon = 'icons/obj/item/landmine/claymore.dmi'
+	icon_state = "m20"
+	var/datum/wires/landmine/claymore/trigger_wire = null
+	var/obj/item/device/assembly/signaler/signaler = null
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/landmines.dm:354]: don't need to declare these as null

File: code/game/objects/items/weapons/traps.dm [hunk 7]
```diff
@@ -689,3 +720,110 @@
...
+	//Select a target zone
+	var/target_zone
+	if(L.lying)
+		target_zone = pick(BP_L_FOOT, BP_R_FOOT, BP_L_LEG, BP_R_LEG, BP_L_HAND, BP_L_ARM, BP_R_HAND, BP_R_ARM)
+	else
+		target_zone = pick(BP_L_FOOT, BP_R_FOOT, BP_L_LEG, BP_R_LEG)
+
+	//Try to apply the damage
+	var/success = FALSE
+	success += L.apply_damage(60, DAMAGE_TOXIN, target_zone, used_weapon = src, armor_pen = activated_armor_penetration)
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/traps.dm:757]: 60 toxin damage is an instant necrotic organ, that's too much and instant death pretty much. Should probably apply toxin reagent instead

File: code/game/objects/items/weapons/traps.dm [hunk 8]
```diff
@@ -689,3 +720,110 @@
...
+		var/obj/item/organ/organ = human.get_organ(target_zone)
+
+		human.visible_message(SPAN_DANGER("\The [human] steps on \the [src]!"),
+								SPAN_WARNING(FONT_LARGE(SPAN_DANGER("You step on \the [src], feel your body fall, and something sharp penetrate your [organ.name]!"))),
+								SPAN_WARNING("<b>You feel your body fall, and something sharp penetrate your [organ.name]!</b>"))
+
+		//If it's a human and not an IPC, apply an infection
+		//We are returning early before this step in case something isn't a human, so this should be fine not to catch borgs/bot/exosuits/whatever
+		if(!isipc(L))
+			organ.germ_level += INFECTION_LEVEL_THREE
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/traps.dm:781]: These infections are extremely powerful and completely immobilize someone within minutes. I think it's too much combined with the toxin

This IPC override should also be at the top, together with vaurca maybe. I don't think carapace or metal would be getting damaged from punji sticks

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/landmines.dm (hunk 3 above)
Comment (FluffyGhoster) [reply code/game/objects/items/weapons/landmines.dm:278]: They cannot keep walking, the proc shows an alert (tgui_alert) that prevents you from moving until you acknowledge it or re-focus the client window, it would also not make much sense that it gives you pain since it's not exploded yet

File: code/game/objects/items/weapons/landmines.dm (hunk 4 above)
Comment (FluffyGhoster) [reply code/game/objects/items/weapons/landmines.dm:284]: It's supposed to convey that you feel something very bad (pressing down the trigger of the mine) under your foot, how would you reword it to make it more clear without it being an obvious "AYO YOU STEPPED ON A MINE!" ?

File: code/game/objects/items/weapons/traps.dm (hunk 7 above)
Comment (FluffyGhoster) [reply code/game/objects/items/weapons/traps.dm]: Used a nice fertilizer injection instead of direct toxin damage, 10 units

File: code/game/objects/items/weapons/traps.dm (hunk 8 above)
Comment (NonQueueingMatt) [reply code/game/objects/items/weapons/traps.dm]: I mean with the spear reasoning most things wouldn't damage an IPC is the issue. As for the carapace idk, I just think it's a better idea to add some species diversity to the game rather than get stuck on stuff like "is a vaurca carapace harder than steel"

File: code/game/objects/items/weapons/landmines.dm (hunk 4 above)
Comment (NonQueueingMatt) [reply code/game/objects/items/weapons/landmines.dm:284]: Just do `You feel your [left/right] foot step down on a button with a click...`, `Uh...`, `Oh...`

**Review by NonQueueingMatt — APPROVED** [review]

**Review by DreamySkrell — APPROVED** [review]
